### ISSUE TITLE
Extract sub_joint URDF items out of the revolute twins

### ISSUE THREAD

**Issue by SylvainBertrand** [ihmcrobotics/simulation-construction-set-2#127]

**Comment by jespersmith** [comment]
URDF has defined a <mimic> tag that should be sufficient to define the revolute twin in code.

```
<?xml version="1.0" ?>
<robot name="RevoluteTwinsRobot" xmlns:xacro="http://ros.org/wiki/xacro">
  <link name="baseLink">
    <visual>
      <geometry>
        <box size="0.025 0.025 0.0125"/>
      </geometry>
      <material>
          <color rgba="0.2 0.7 0.9 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0.1 0 0.2"/>
    </visual>
  </link>
  <link name="endEffectorLink">
    <inertial>
      <mass value="5"/>
      <origin rpy="0 0 0" xyz="0  0  0.1"/>
      <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.005"/>
    </inertial>
    <visual>
      <geometry>
        <box size="0.0125 0.0125 0.2"/>
      </geometry>
      <material>
          <color rgba="0.2 0.7 0.9 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0 0 0.1"/>
    </visual>
  </link>
    <joint name="jointA" type="revolute">
      <origin rpy="0 0 0" xyz="0.1 0 0.2"/>
      <parent link="baseLink"/>
      <axis xyz="0 1 0"/>
      <child link="bodyAB"/>
      <limit effort="190" lower="-1.1" upper="0.4141" velocity="5.89"/>
   </joint>
    <joint name="jointB" type="revolute">
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <parent link="bodyAB"/>
      <child link="endEffectorLink"/>
      <limit effort="190" lower="-1.1" upper="0.4141" velocity="5.89"/>
      <mimic joint="jointA" multiplier="1.25" offset="0">
      </mimic>
    </joint>
    <link name="bodyAB">
      <inertial>
        <mass value="0.025"/>
        <origin rpy="0 0 0" xyz="0  0  0.1"/>
        <inertia ixx="1.0e-8" ixy="0" ixz="0" iyy="1.0e-8" iyz="0" izz="1.0e-8"/>
      </inertial>
      <visual>
        <geometry>
          <cylinder radius="0.0125" length="0.2"/>
        </geometry>
        <material>
            <color rgba="0.92 0.85 0.2 1"/>
        </material>
        <origin rpy="0 0 0" xyz="0  0  0.1"/>
      </visual>
    </link>
</robot>
```

**Comment by SylvainBertrand** [comment]
Alright so I created a new branch `feature/revolute-twins-improvement` which adds 2 new options for defining a revolute twins in the URDF file:
- Minimalistic: this corresponds to the URDF you posted just above. There is 1 minor with this: how to name the revolute twins joint. For now, I did a naive implementation which names it `jointA.getName() + "_" + jointB.getName()`. Here's the URDF sample: https://github.com/ihmcrobotics/simulation-construction-set-2/blob/feature/revolute-twins-improvement/scs2-examples/src/main/resources/urdf/SimpleRevoluteTwinsRobotTypeB.urdf
- Halfway between original and minimalistic: we kind of talked about it briefly, it's the same as the minimalistic + partial declaration of the revolute twins joints with: name, parent and child links. The parser will retrieve the sub-joints. Here's the URDF sample: https://github.com/ihmcrobotics/simulation-construction-set-2/blob/feature/revolute-twins-improvement/scs2-examples/src/main/resources/urdf/SimpleRevoluteTwinsRobotTypeC.urdf

Diff with develop:
https://github.com/ihmcrobotics/simulation-construction-set-2/compare/develop...feature/revolute-twins-improvement

Lemme know what you think.
